qiskit_machine_learning/runtime/hookbase.py

```diff
@@ -0,0 +1,58 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""HookBase for the Torch Runtime"""
+
+from typing import Optional, Any
+
+
+class HookBase:
+    """Base class for hooks that can be registered in ``Trainer``.
+    A hook can implement 6 methods. Each method is called before/after the corresponding processes.
+    """
```

Comment:
So this class is essentially a set of callback functions - and its serialized and executed on the server side? Are there restrictions on what one can do in these methods in any way - e.g. if I try and call other code I have on my local machine I guess it will fail, or are all its dependents serialized over too as needed?

So these methods (what I look at as callbacks) don;t have any additional parameters - e.f before/after step I have no info about the step itself just thats its called before hand. Maybe as an end user I need to create some subclass and store references to objects I can query about the step or whatever else it is that is applicable etc such that then before/after step have some meaning I can get to.

Maybe this will become clear from tutorial? Would it make sense for the docs here to show example usages or do you think that will all be done by some tutorial notebook.

Reply:
>So this class is essentially a set of callback functions
Exactly! I was planning to write the detail information in the tutorial. However, I agree with that the docstring s does not explain enough. I will add more information in the docstrings as well.